File: Queue/ReverseKPart/reverse.py
class Solution:
    def reverseFirstK(self, q, k):
        if k <= 1:
            return q

        n = len(q)

        if k > n:
            return q

        st = []

        for _ in range(k):
            st.append(q.popleft())

        while st:
            q.append(st.pop())

        for _ in range(n - k):
            q.append(q.popleft())

        return q

File: Queue/ReverseKPart/reverse.cpp
#include <queue>
#include <stack>
using namespace std;

class Solution {
  public:
    queue<int> reverseFirstK(queue<int> q, int k) {
        // code here
        if(k<=1)return q;
        stack<int> st;
        int n = q.size();
        if(k>n)return q;
        int count = k;
        while(count){
            int val = q.front();
            st.push(val);
            q.pop();
            count--;
        }
        while(!st.empty()){
            int val = st.top();
            q.push(val);
            st.pop();
        }
        count = n-k;
        while(count){
            int val = q.front();
            q.push(val);
            q.pop();
            count--;
        }
        return q;
    }
};